streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -172,6 +170,11 @@ public void openDB(ProcessorContext context) {
             valueSerde == null ? (Serde<V>) context.valueSerde() : valueSerde);
 
         this.dbDir = new File(new File(context.stateDir(), parentDir), this.name);
+
+        if (!hasPreExistingSstFiles && prepareForBulkload) {
+            options.prepareForBulkLoad();
```

Comment:
This is a general comment: thinking about this more: since we are more likely to have existing sstFiles when restoration is needed, this fix is effectively voiding the optimization we did for bulk loading. I'm wondering if we can overwrite the `num_levels` config after setting `options.prepareForBulkload` so that we can still use bulk loading even with existing sst files? Looking at the rocksDB doc it seems this option is not only design for loading into an empty store?